flight/search/cell_map.py

```diff
@@ -0,0 +1,140 @@
+
+        Parameters
+        ----------
+        points : list[list[tuple[float, float] | str]]
+            the points that define the CellMap's data
+
+        Returns
+        -------
+        num_valids : int
+            the number of valid cells in the cell map
+        """
+
+        count: int = 0
+        i: int
+        j: int
+        for i, _ in enumerate(points):
+            for j, _ in enumerate(points[0]):
+                if points[i][j] != "X":
+                    count += 1
+        return count
+
+    def __init_map(
+        self, points: list[list[tuple[float, float] | str]], odlc_count: int
+    ) -> list[list[Cell]]:
+        """
+        This method creates the two-dimensional array filled with Cell
+        objects used by the CellMap.
+
+        Parameters
+        ----------
+        points : list[list[tuple[float, float] | str]]
+            A two-dimensional array of latitude and longitude points that defines
+            the search area
+        odlc_count : int
+            the number of ODLCs in the search area
+
+        Returns
+        -------
+        final_map : list[list[Cell]]
+            the two-dimensional list of cell objects
+        """
+        r_list: list[list[Cell]] = []
+        i: int
+        j: int
+        for i, _ in enumerate(points):
+            row: list[Cell] = []
+            for j, _ in enumerate(points[0]):
+                if points[i][j] != "X":  # ensures it is not the only used string value
+                    row.append(
+                        Cell(
+                            odlc_count / self.num_valids,
+                            False,
+                            points[i][j][0],  # type: ignore
+                            points[i][j][1],  # type: ignore
+                            True,
+                        )
+                    )
+                else:
+                    row.append(Cell(0, False, -1.0, -1.0, False))
+            r_list.append(row)
+        return r_list
+
+    def __getitem__(self, index: int) -> list[Cell]:
+        return self.data[index]
+
+    def display(self) -> None:
+        """
+        Prints out the current CellMap in the standard output.
+        """
+        i: int
+        j: int
+        for i, _ in enumerate(self.data):
+            row_string: str = ""
+            for j, _ in enumerate(self.data[0]):
+                if not self.data[i][j].is_valid:
+                    row_string += " "
+                else:
+                    row_string += "X"
+            print(row_string)
+
+    def __init__(self, points: list[list[tuple[float, float] | str]], odlc_count: int = 1) -> None:
+        """
+        initializes the object with given parameters
+
+        Parameters
+        ----------
+        points : list[list[tuple[float, float] | str]]
+            a collection of latitude, longitude coordinates to define the map
+        odlc_count : int
+            the number of ODLCs in the flight area
+        """
```

Comment:
Your `__init__` function should be the first function in the class. Move this above all other functions in the `CellMap` class.